nano/node/active_transactions.cpp

```diff
@@ -644,22 +645,102 @@ void nano::active_transactions::update_difficulty (std::shared_ptr<nano::block>
 	auto existing_election (roots.get<tag_root> ().find (block_a->qualified_root ()));
 	if (existing_election != roots.get<tag_root> ().end ())
 	{
-		auto difficulty (block_a->difficulty ());
-		if (difficulty > existing_election->difficulty)
+		double multiplier (normalized_multiplier (block_a, existing_election->election->status.winner));
+		if (multiplier > existing_election->multiplier)
 		{
 			if (node.config.logging.active_update_logging ())
 			{
-				node.logger.try_log (boost::str (boost::format ("Block %1% was updated from difficulty %2% to %3%") % block_a->hash ().to_string () % nano::to_string_hex (existing_election->difficulty) % nano::to_string_hex (difficulty)));
+				node.logger.try_log (boost::str (boost::format ("Block %1% was updated from multiplier %2% to %3%") % block_a->hash ().to_string () % existing_election->multiplier % multiplier));
 			}
-			roots.get<tag_root> ().modify (existing_election, [difficulty](nano::conflict_info & info_a) {
-				info_a.difficulty = difficulty;
+			roots.get<tag_root> ().modify (existing_election, [multiplier](nano::conflict_info & info_a) {
+				info_a.multiplier = multiplier;
 			});
 			existing_election->election->publish (block_a);
 			add_adjust_difficulty (block_a->hash ());
 		}
 	}
 }
 
+double nano::active_transactions::normalized_multiplier (std::shared_ptr<nano::block> block_a, std::shared_ptr<nano::block> winner_a)
+{
+	auto difficulty (block_a->difficulty ());
+	uint64_t threshold (0);
+	if (block_a->has_sideband ())
+	{
+		threshold = nano::work_threshold (block_a->work_version (), block_a->sideband ().details);
+	}
+	else if (winner_a != nullptr && winner_a->has_sideband () && winner_a->hash () == block_a->hash ())
+	{
+		threshold = nano::work_threshold (block_a->work_version (), winner_a->sideband ().details);
+	}
+	else
+	{
+		threshold = nano::work_threshold_base (block_a->work_version ());
+	}
+	double multiplier (nano::difficulty::to_multiplier (difficulty, threshold));
+	debug_assert (multiplier >= 1);
+	if (multiplier >= 1)
+	{
+		multiplier_normalization (multiplier, threshold);
+	}
+	return multiplier;
+}
+
+void nano::active_transactions::multiplier_normalization (double & multiplier_a, uint64_t const threshold_a)
```

Comment:
I think this would work better as an overload of `normalized_multiplier` with different parameter types and returning the new multiplier instead of modifying the parameter.

Reply:
Thanks